_posts/2022-12-01-trino-summit-2022-sk-telecom-recap.md

```diff
@@ -0,0 +1,98 @@
+
+As Trino adoption grew within the company to the tune of 300 requests per
+minute, they eventually faced challenges scaling. Not only were the number of
+requests growing, but users increased the range of data they wanted to query.
+The total data size they were evaluating was petabytes in size, involving
+terabyte size query input. This was all bing processed over hundreds of nodes.
+Many users were queued in a blocked state while waiting for resources to become
+available. At this point, Trino query performance also began to suffer. It was
+at this point the data engineering team began investigating how they could both
+scale and improve individual query performance.
+
+To find the root cause, they had to expand beyond the web UI. They began
+collecting all result json files, coordinator and worker jmx stats, system
+metrics, and Trino logs to build out their own metrics dashboard. The two main
+causes was that input data was too large, and they saw spikes in the number of
+`BlockedSplit` operations that lead to queries waiting for other tasks to
+complete and in a blocked state. They initially aimed to address this by
+updating some settings to increase thread counts and tuning queries, but this
+still didn't achieve the desired results. The ultimate bottleneck was the Hive
+metastore and the expensive list operations that caused many of the blocked
+operations to finish slowly.
+
+At this point, the team reevaluated their needs to consider alternative solution
+spaces. They needed a better indexing strategy on the data Hive data with a
+flexible partitioning strategy. They also needed to remove the bottleneck on the
+metadata for this data while still maintaining compatibility across multiple
+query engines as Hive did for the most part.
+
+The team looked at the existing set of novel data lake connectors available in
+Trino version 356, which at the time only included Iceberg. SK Telecom was 
+immediately impressed by the metadata indexing in the Iceberg project. They 
+particularly liked the snapshot isolation as new data was created or existing 
+data is modified. They were able to speed up queries using data file pruning
+using partition and column stats, stored in the manifest files in the object
+files themselves. This resulted in a huge decrease in the input data size of the
+queries.
+
+After running a benchmark, they found Iceberg reduced the input size on the
+order of hundreds down to under ten gigabytes. They also investigated adding a
+high amount of partitions to continue lowering the input data. They found that
+there's a tradeoff where too much partitions begins to increase query planning
+time, so they found a sweet spot where the input data size was around six
+gigabytes and planning only took 70 milliseconds.
+
+This is just the tip of the iceberg of all the takeaways from this talk on how
+Iceberg helped SK Telecom addressed their scalability issues on their platform.
```

Comment:
Suggest rewriting to "This summary is just the tip of the iceberg of all the information JaeChang and Jennifer shared with us about how Iceberg helped SK Telecom with their Trino scaling issues."